### PR TITLE
Add SkipCRDs to CR

### PR DESCRIPTION
**Pull request by yasn77**
Towards giantswarm/giantswarm#13746

Pass on to helmclient so it can decide whether to install release CRDs

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by yasn77** [comment]
@rossf7 Tests failing as helmclient hasn't been released with latest changes.

**Comment by rossf7** [comment]
> Tests failing as helmclient hasn't been released with latest changes.

@yasn77 We should release helmclient v4.2.0 Can you do that using the release PR automation?

But you can also use the master branch in the PR with

```
go get github.com/giantswarm/helmclient/v4@master
go mod tidy
```

**Comment by rossf7** [comment]
Actually the build is failing due to apiextensions. This was updated by dependabot yesterday https://github.com/giantswarm/chart-operator/pull/682 so this also needs syncing with master.

**Comment by yasn77** [comment]
@rossf7 Not sure what to do with the failed build :/

**Comment by rossf7** [comment]
Damn that's a pain. Welcome to hell. Dependency edition :(

```
# sigs.k8s.io/cluster-api/util/predicates
/go/pkg/mod/github.com/giantswarm/cluster-api@v0.3.10-gs/util/predicates/generic_predicates.go:147:91: e.MetaNew undefined (type event.UpdateEvent has no field or method MetaNew)
/go/pkg/mod/github.com/giantswarm/cluster-api@v0.3.10-gs/util/predicates/generic_predicates.go:150:88: e.Meta undefined (type event.CreateEvent has no field or method Meta)
/go/pkg/mod/github.com/giantswarm/cluster-api@v0.3.10-gs/util/predicates/generic_predicates.go:153:88: e.Meta undefined (type event.DeleteEvent has no field or method Meta)
/go/pkg/mod/github.com/giantswarm/cluster-api@v0.3.10-gs/util/predicates/generic_predicates.go:156:89: e.Meta undefined (type event.GenericEvent has no field or method Meta)
```

I think this is related to our forks of cluster-api and controller-runtime, We bumped controller-runtime from 0.6.3 to 0.7.0 via dependabot https://github.com/giantswarm/helmclient/pull/308 but I think thats broke it.

I had a similar problem in app-operator last week. I think we should downgrade helmclient to 0.6.5. https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.6.5

Do you want to have a go at that? I'm also fine to look at it tomorrow? As its not really related to your change.

**Comment by yasn77** [comment]
> Damn that's a pain. Welcome to hell. Dependency edition :(
> 
> ```
> # sigs.k8s.io/cluster-api/util/predicates
> /go/pkg/mod/github.com/giantswarm/cluster-api@v0.3.10-gs/util/predicates/generic_predicates.go:147:91: e.MetaNew undefined (type event.UpdateEvent has no field or method MetaNew)
> /go/pkg/mod/github.com/giantswarm/cluster-api@v0.3.10-gs/util/predicates/generic_predicates.go:150:88: e.Meta undefined (type event.CreateEvent has no field or method Meta)
> /go/pkg/mod/github.com/giantswarm/cluster-api@v0.3.10-gs/util/predicates/generic_predicates.go:153:88: e.Meta undefined (type event.DeleteEvent has no field or method Meta)
> /go/pkg/mod/github.com/giantswarm/cluster-api@v0.3.10-gs/util/predicates/generic_predicates.go:156:89: e.Meta undefined (type event.GenericEvent has no field or method Meta)
> ```
> 
> I think this is related to our forks of cluster-api and controller-runtime, We bumped controller-runtime from 0.6.3 to 0.7.0 via dependabot [giantswarm/helmclient#308](https://github.com/giantswarm/helmclient/pull/308) but I think thats broke it.
> 
> I had a similar problem in app-operator last week. I think we should downgrade helmclient to 0.6.5. https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.6.5
> 
> Do you want to have a go at that? I'm also fine to look at it tomorrow? As its not really related to your change.

A great introduction to Go modules ;)

@rossf7  PR Opened https://github.com/giantswarm/helmclient/pull/318

**Comment by rossf7** [comment]
> A great introduction to Go modules ;)

Go modules works pretty well but k8s client-go pushes it to breaking point sometimes :(

The problem is in k8s 1.18 the context params got added which is a breaking change. But the k8s version is v0.19.4 and 0.X is not treated as a breaking change. This is why we need the forks of CAPI.

Could you test with the master branch of helmclient before cutting a new release? Just to be sure this does actually fix it.

**Comment by rossf7** [comment]
> Could you test with the master branch of helmclient before cutting a new release? Just to be sure this does actually fix it.

@yasn77 Sorry don't worry about this. I just did a quick test and its fine.

File: go.mod [hunk 1]
```diff
@@ -26,6 +26,7 @@ require (
 	k8s.io/api v0.19.4
 	k8s.io/apimachinery v0.19.4
 	k8s.io/client-go v0.19.4
+	sigs.k8s.io/controller-runtime v0.7.0 // indirect
```
Comment (rossf7) [inline go.mod:29]: Something is still wrong and 0.7.0 won't compile.. Can you try deleting this line and run `go mod tidy`?

Comment (yasn77) [reply go.mod]: Done and pushed

File: service/controller/chart/resource/release/create.go [hunk 2]
```diff
@@ -82,8 +83,12 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 	// If we do timeout the install will continue in the background.
 	// We will check the progress in the next reconciliation loop.
 	go func() {
+		if skipCRDs {
+			r.logger.Debugf(ctx, "helm release %#q has SkipCRDs set to true, not installing release CRDs", releaseState.Name)
```
Comment (rossf7) [inline service/controller/chart/resource/release/create.go:87]:
```suggestion
			r.logger.Debugf(ctx, "helm release %#q has SkipCRDs set to true, not installing CRDs", releaseState.Name)
```

I think we can drop the 2nd release. WDYT?

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,10 @@ and this project's packages adheres to [Semantic Versioning](http://semver.org/s
 
 ## [Unreleased]
 
+### Added
+
+- Add SkipCRDs to CR, this will be passed on to helmclient
```
Comment (rossf7) [inline CHANGELOG.md:12]:
```suggestion
- Add support for skip CRD flag when installing Helm releases.
```

**Review by rossf7 — APPROVED** [review]
Thx! LGTM

Just a couple of 🇬🇧 nits.

**Pull request merged**